store/store.go

```diff
@@ -930,6 +949,48 @@ func (s *Store) Find(search *Search, user *auth.UserState) ([]*snap.Info, error)
 	return snaps, nil
 }
 
+// Sections retrieves the list of available store sections.
+func (s *Store) Sections(user *auth.UserState) ([]string, error) {
+	u := *s.sectionsURI // make a copy, so we can mutate it
+
+	q := u.Query()
+
+	q.Set("confinement", "strict")
+
+	u.RawQuery = q.Encode()
+
+	reqOptions := &requestOptions{
+		Method: "GET",
+		URL:    &u,
+		Accept: halJsonContentType,
+	}
+	resp, err := s.doRequest(s.client, reqOptions, user)
```

Comment:
as this is new code, could you write it already doing the retry dance? Search for `retry.Start` in this same file for examples. We're slowly rewriting all things that call `doRequest` to work this way, but we shouldn't add to the backlog.